dask/array/core.py

```diff
@@ -1141,7 +1149,21 @@ def __getitem__(self, index):
         dsk, chunks = slice_array(out, self.name, self.chunks, index2)
 
         graph = HighLevelGraph.from_collections(out, dsk, dependencies=[self])
-        return Array(graph, out, chunks, dtype=self.dtype)
+        if isinstance(index2, tuple):
+            new_index = []
+            for i in range(len(index2)):
+                if not isinstance(index2[i], tuple):
+                    types = [int, np.int_, list, np.ndarray]
+                    cond = any([isinstance(index2[i], t) for t in types])
+                    new_index.append(slice(0, 0) if cond else index2[i])
+                else:
+                    new_index.append(tuple([Ellipsis if i is not None else None for i in index2]))
+            new_index = tuple(new_index)
+            meta = self._meta[new_index].astype(self.dtype)
+        else:
+            meta = self._meta[index2].astype(self.dtype)
```

Comment:
I'm going to try to defer to @jakirkham on this :)

Reply:
Sorry I'm not totally following what is happening in this block. Could you please provide a summary, @pentschev?

Reply:
Sure, what happens here is very simple, basically creating a `new_index` in this order:
1. Check all elements of the `index2` tuple, if it's not a tuple, make it a `slice(0, 0)`;
1. For elements that are not tuples (e.g., a single index), make it an `Ellipsis` if the index is valid, otherwise make it `None`.

This way, I convert all cases to 0-sized slices. For example:
- `[1, slice(1:2), :]` becomes shape `(0, 0, ...)`
- `[None, 1, None]` becomes shape `(1, 0, 1)`, the `1`s that arise from None should probably still be fixed, even though it may not be a problem generally

Reply:
Thanks for providing more info. I'm still confused unfortunately. Have a couple of follow-up questions.

What is the difference between using `slice(0, 0)` and `Ellipsis` on `_meta`?

Also, I thought `None` would add singleton dimensions. Why do we want that for this use case?

Reply:
> What is the difference between using `slice(0, 0)` and `Ellipsis` on `_meta`?

I think they're equivalent, I probably started using `Ellipsis` and forgot to fix that. Plus, having two consecutive `Ellipsis` isn't allowed, so that's wrong for sure, I'll fix it.

> Also, I thought `None` would add singleton dimensions. Why do we want that for this use case?

It does add a singleton dimension. We would want one if the user requests so, as per @mrocklin's previous example [here](https://github.com/dask/dask/pull/4543#discussion_r263138930).

Just to be clear here, I was attempting to cover all cases I could think of. Maybe not all of them would make sense in a realistic context, but are technically allowed, so we should cover them.